Dockerfile.pr-builder

```diff
@@ -0,0 +1,60 @@
+FROM continuumio/miniconda
+MAINTAINER Hail Team <hail@broadinstitute.org>
+
+USER root
+RUN apt-get -y install \
+    bash \
+    cmake \
+    curl \
+    g++ \
+    git \
+    openjdk-8-jdk-headless \
+    r-base \
+    tar \
+    unzip
+
+RUN apt-get -y install libnlopt-dev # https://github.com/jyypma/nloptr/issues/40
+RUN Rscript -e 'install.packages(c("jsonlite", "SKAT", "logistf"), repos = "http://cran.us.r-project.org")'
+RUN apt-get -y install r-cran-ncdf4 r-cran-rnetcdf libcurl4-openssl-dev libxml2-dev
+RUN Rscript -e 'source("https://bioconductor.org/biocLite.R"); biocLite("GENESIS"); biocLite("SNPRelate"); biocLite("GWASTools")'
+
+WORKDIR /plink
+ADD http://www.cog-genomics.org/static/bin/plink/plink_linux_x86_64.zip /plink
+RUN unzip plink_linux_x86_64.zip && \
+    rm -rf plink_linux_x86_64.zip && \
+    ln -s /plink/plink /bin/plink
+
+WORKDIR /qctool
+ADD http://www.well.ox.ac.uk/~gav/resources/qctool_v2.0.1-CentOS6.8-x86_64.tgz /qctool
+RUN tar --strip-components=1 -xvzf qctool_v2.0.1-CentOS6.8-x86_64.tgz -C /qctool && \
+    rm -rf qctool_v2.0.1-CentOS6.8-x86_64.tgz && \
+    ln -s /qctool/qctool /bin/qctool
+
+# cache (almost all) gradle dependencies (and gradle itself)
+#
+# Some plugin we use hides its dependencies in a so-called "detached
+# configuration" so there is no way for us to ensure it gets cached.
+WORKDIR /hail
+COPY gradlew build.gradle settings.gradle ./
```

Comment:
If these are the only files we need to copy into the image, it seems like overkill to build the image in `hail/` and ignore almost everything in the `.dockerignore`. Right now, all of the hail source is getting tarred up and copied into the docker daemon whenever you build the image, but most of that isn't used. We could just ignore everything but these files, but it seems better to put the Dockerfile in its own directory with copies of just the files it needs. Then the file could just be called `Dockerfile` too.